src/topology/continuous_function/t0_sierpinski.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2022 Ivan Sadofschi Costa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ivan Sadofschi Costa
+-/
+import topology.order
+import topology.continuous_function.basic
+
+/-!
+# Any T0 space embeds in a product of copies of the Sierpinski space
+
+We consider `Prop` with the Sierpinski topology. If `t` is a topology on `X`, there is a continuous
+map `product_of_mem_opens : X → Π u ∈ t, Prop` defined as the product of the maps `X → Prop` defined
+by `x ↦ x ∈ u`.
+
+The map `product_of_mem_opens` is always inducing. Whenever `X` is T0, `product_of_mem_opens` is
+also injective and therefore an embedding.
+-/
+
+noncomputable theory
+
+variables {X : Type*} [t : topological_space X] [t0 : t0_space X]
+
+instance : topological_space (Π (u : {s : set X | t.is_open s}), Prop) :=
```

Comment:
Is there any reason to use `t.is_open` instead of `is_open`?

Reply:
I didn't know about `opens`, thanks!